packages/backend-for-frontend/src/services/agreementService.ts

```diff
@@ -112,6 +110,51 @@ export function agreementServiceBuilder(
             showOnlyUpgradeable,
             eservicesIds,
             consumersIds,
+            states,
+          },
+          headers: ctx.headers,
+        });
+
+      const agreements = results.map((a) =>
+        enrichAgreementListEntry(a, clients, ctx)
+      );
+      return {
+        pagination: {
+          limit,
+          offset,
+          totalCount,
+        },
+        results: await Promise.all(agreements),
+      };
+    },
+
+    async getProducersAgreements(
+      {
+        offset,
+        limit,
+        eservicesIds,
+        producersIds,
+        states,
+        showOnlyUpgradeable,
+      }: {
+        offset: number;
+        limit: number;
+        eservicesIds: string[];
+        producersIds: string[];
+        states: bffApi.AgreementState[];
+        showOnlyUpgradeable?: boolean;
+      },
+      ctx: WithLogger<BffAppContext>
+    ): Promise<bffApi.Agreements> {
+      ctx.logger.info("Retrieving agreements");
+
+      const { results, totalCount } =
+        await agreementProcessClient.getAgreements({
+          queries: {
+            offset,
+            limit,
+            showOnlyUpgradeable,
+            eservicesIds,
             producersIds,
             states,
           },
```

Comment:
See comments above: here we should fix the producer to the requester and leave the consumer filter customizable by the front-end. Something like:

```suggestion
        await agreementProcessClient.getAgreements({
          queries: {
            producersIds: [ctx.authData.organizationId],
            offset,
            limit,
            showOnlyUpgradeable,
            eservicesIds,
            consumerIds,
            states,
          },
```